extensions/resteasy-classic/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientBase.java

```diff
@@ -285,9 +374,29 @@ private void configureBaseUrl(RestClientBuilder builder) {
                 : config.getOptionalValue(String.format(propertyFormat, propertyPrefix), type);
     }
 
+    @SuppressWarnings("SameParameterValue")
     private <T> Optional<T> getOptionalProperty(String propertyName, Class<T> type) {
         final Config config = ConfigProvider.getConfig();
         return config.getOptionalValue(propertyName, type);
     }
 
+    private <PropertyType> Optional<PropertyType> getClientConfigPropertyWithFallback(
```

Comment:
I think that any fallback / relocate of configuration properties should use the SR Config APIs for that. In that way, any lookup using the Config API will apply the relocate / fallback instead of just doing it here. You can check a few examples here:

https://smallrye.io/docs/smallrye-config/interceptors/interceptors.html#fallback-interceptor
https://github.com/quarkusio/quarkus/blob/abb7978101f98ebfc8eba201140395347f526137/extensions/smallrye-opentracing/runtime/src/main/java/io/quarkus/smallrye/opentracing/runtime/TracingConfigRelocateConfigSourceInterceptor.java

Reply:
Done, see `RestClientFallbackConfigSourceInterceptor`.